plugins/woocommerce/client/blocks/assets/js/blocks/product-collection/edit/inspector-controls/taxonomy-controls/index.tsx

```diff
@@ -36,66 +40,125 @@ export const useTaxonomies = (): Taxonomy[] => {
 	}, [ taxonomies ] );
 };
 
-function TaxonomyControls( {
+/**
+ * Normalize the name so first letter of every word is capitalized.
+ */
+const normalizeName = ( name: string | undefined | null ) => {
+	if ( ! name ) {
+		return '';
+	}
+
+	return name
+		.split( ' ' )
+		.map( ( word ) => word.charAt( 0 ).toUpperCase() + word.slice( 1 ) )
+		.join( ' ' );
+};
+
+/**
+ * Shared hook for taxonomy control logic - filters taxonomies based on context and provides common handlers.
+ */
+function useTaxonomyControls( {
 	setQueryAttribute,
 	trackInteraction,
 	query,
-}: QueryControlProps ) {
+	collection,
+	isFiltersPanel,
+}: QueryControlProps & { collection: string | undefined } & {
+	isFiltersPanel?: boolean;
+} ) {
 	const { taxQuery } = query;
-
 	const taxonomies = useTaxonomies();
-	if ( ! taxonomies || taxonomies.length === 0 ) {
-		return null;
-	}
 
-	/**
-	 * Normalize the name so first letter of every word is capitalized.
-	 */
-	const normalizeName = ( name: string | undefined | null ) => {
-		if ( ! name ) {
-			return '';
+	const filteredTaxonomies = useMemo( () => {
+		if ( ! taxonomies || taxonomies.length === 0 ) {
+			return [];
 		}
 
-		return name
-			.split( ' ' )
-			.map( ( word ) => word.charAt( 0 ).toUpperCase() + word.slice( 1 ) )
-			.join( ' ' );
+		if ( collection === CoreCollectionNames.BY_CATEGORY ) {
+			return taxonomies.filter( ( taxonomy ) =>
+				isFiltersPanel
+					? taxonomy.slug !== 'product_cat'
+					: taxonomy.slug === 'product_cat'
+			);
+		}
+		if ( collection === CoreCollectionNames.BY_TAG ) {
+			return taxonomies.filter( ( taxonomy ) =>
+				isFiltersPanel
+					? taxonomy.slug !== 'product_tag'
+					: taxonomy.slug === 'product_tag'
+			);
+		}
+
+		return isFiltersPanel ? taxonomies : [];
+	}, [ taxonomies, collection, isFiltersPanel ] );
+
+	const createHandleChange = ( slug: string ) => ( newTermIds: number[] ) => {
+		setQueryAttribute( {
+			taxQuery: {
+				...taxQuery,
+				[ slug ]: newTermIds,
+			},
+		} );
+		trackInteraction( `${ CoreFilterNames.TAXONOMY }__${ slug }` );
+	};
+
+	const createDeselectCallback = ( slug: string ) => () => {
+		createHandleChange( slug )( [] );
 	};
 
+	const shouldShowTaxonomyControl =
+		taxonomies && taxonomies.length > 0 && filteredTaxonomies.length > 0;
```

Comment:
Feels like this can be simplified to:

```suggestion
	const shouldShowTaxonomyControl = filteredTaxonomies.length > 0;
```

This is the only condition that eventually matters 🤔 WDYT?

Reply:
Yup, updated it now!